src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -236,6 +239,16 @@ public Expression add(Operator op, ByteBuffer value)
                 lower = new Bound(value, indexTermType, true);
                 upper = lower;
                 break;
+
+            case IN:
+                if (indexTermType.isValid(value))
```

Comment:
This might pass for the string case, but I don't think we want to pass the `value` BB directly to `isValid()`. I think you would need to do something similar to what you've done in `termMatches()` to get the individual items and pass those. Having said that, I'm not sure we need to validate here at all if we'll fail somewhere further up the stack with invalid `IN` values. If we avoid validation, we can also just reuse the `EQ`/`CONTAINS`/`CONTAINS KEY` block above...

Reply:
Indeed...trying to supply something that isn't text will fail during prepare...

```
Caused by: org.apache.cassandra.exceptions.InvalidRequestException: Invalid INTEGER constant (11) for "k3" of type text
	at org.apache.cassandra.cql3.terms.Constants$Literal.prepare(Constants.java:270)
	at org.apache.cassandra.cql3.terms.Constants$Literal.prepare(Constants.java:218)
	at org.apache.cassandra.cql3.terms.Terms$Raw$2.prepare(Terms.java:316)
	at org.apache.cassandra.cql3.Relation.toRestriction(Relation.java:216)
	at org.apache.cassandra.cql3.restrictions.StatementRestrictions.<init>(StatementRestrictions.java:218)
```